Sluggo-SwiftUI/Models/Managers/UserManager.swift

```diff
@@ -10,41 +10,56 @@ import Foundation
 class UserManager {
     static let urlBase = "auth/"
     private var identity: AppIdentity
+    private let requestLoader: CanNetworkRequest
 
-    init(identity: AppIdentity) {
+    init(identity: AppIdentity, requestLoader: CanNetworkRequest = JsonLoader()) {
         self.identity = identity
+        self.requestLoader = requestLoader
     }
 
-    public func getUser(completitonHandler: @escaping(Result<AuthRecord, Error>) -> Void) {
+    public func getUser() async -> Result<AuthRecord, Error> {
         let requestBuilder = URLRequestBuilder(url: URL(string: identity.baseAddress + "auth/user/")!)
             .setMethod(method: .GET)
             .setIdentity(identity: self.identity)
 
-        JsonLoader.executeCodableRequest(request: requestBuilder.getRequest(), completionHandler: completitonHandler)
+        return await requestLoader.executeCodableRequest(request: requestBuilder.getRequest())
     }
 
     public func doLogin(username: String,
-                        password: String, completionHandler: @escaping(Result<TokenRecord, Error>) -> Void) {
+                        password: String) async -> Result<TokenRecord, Error> {
         let params = ["username": username, "password": password] as [String: String]
         guard let body = try? JSONSerialization.data(withJSONObject: params, options: []) else {
-            completionHandler(.failure(Exception.runtimeError(message:
-                                                                "Failed to serialize JSON for doLogin in UserManager")))
-            return
+            let errorMessage = "Failed to serialize JSON for doLogin in UserManager"
+            return .failure(Exception.runtimeError(message: errorMessage))
         }
 
         let requestBuilder = URLRequestBuilder(url: URL(string: identity.baseAddress + UserManager.urlBase + "login/")!)
             .setData(data: body)
             .setMethod(method: .POST)
 
-        JsonLoader.executeCodableRequest(request: requestBuilder.getRequest(), completionHandler: completionHandler)
+        return await requestLoader.executeCodableRequest(request: requestBuilder.getRequest())
+    }
+    
+    public func doRefresh() async -> Result<RefreshRecord, Error> {
```

Comment:
Not sure if it makes the most sense hierarchically for doRefresh to be in the UserManager. It also looks like this function isn't being used yet. I'm not sure on the details on how networking works in iOS but it seems like this function should exist within the JsonLoader class, since it should be called whenever any authenticated request fails requiring a refresh for that token.